docs/tutorials/build-files-archive-with-dal.md

```diff
@@ -0,0 +1,688 @@
+   $\mathrm{page\_size}$ in our case (since it is the size of the buffer).
+
+Passing the DAL parameters as an argument for our function requires to import
+the type, which is not correctly exported by the `tezos-smart-rollup` crate as
+of version 0.2.0. This means we need to modify `Cargo.toml` to have direct
+access to `tezos-smart-rollup-host`.
+
+```diff
+ [dependencies]
+ tezos-smart-rollup = { version = "0.2.2", features = [ "proto-alpha" ] }
++tezos-smart-rollup-host = { version = "0.2.2", features = [ "proto-alpha" ] }
+```
+
+Repeating the commands to originate and run our Smart Rollup, we get the
+following result after a while.
+
+```
+...
+RollupDalParameters { number_of_slots: 32, attestation_lag: 4, slot_size: 65536, page_size: 4096 }
+No attested slot at index 0 for level 56875
+See you in the next level
+RollupDalParameters { number_of_slots: 32, attestation_lag: 4, slot_size: 65536, page_size: 4096 }
+Attested slot at index 0 for level 56876: [16, 0, 0, 2, 89, 87, 0, 0, 0, 0]
+See you in the next level
+RollupDalParameters { number_of_slots: 32, attestation_lag: 4, slot_size: 65536, page_size: 4096 }
+No attested slot at index 0 for level 56877
+See you in the next level
+...
+```
+
+But at first, you are likely to be confused by the logs of the kernel. For the
+first 4 Tezos blocks produced after the origination of the Smart Rollup, the
+kernel will report that no slot has been attested for the targeted level, _even
+if Explorus states the opposite_. This is because, as of January, 2024, a Smart
+Rollup cannot fetch the content of a slot published before it is originated.
+This is why you have to wait for 4 blocks before seeing slot page contents being
+logged.
+
+### Task 3. Publishing on the DAL
+
+Fetching slots produced by the Tezos core developers in order to verify that the
+DAL works as expected on Weeklynet is one thing, but we will not stop there.
+The next task for us is to publish data of our own, and to verify our kernel is
+indeed able to fetch it.
+
+We will use the slot index 10 for this experiment.
+
+:::note Planning ahead
+Before trying to run the code yourself, have a look at Explorus and choose a
+slot that is not currently being used.
+:::
+
+Kill your DAL node, and restart it with a new `--producer-profiles` argument.
+
+```bash
+octez-dal-node --endpoint https://rpc.weeklynet-2024-01-03.teztnets.com \
+    --producer-profiles=10 --data-dir _dal_node
+```
+
+Then, modify your kernel to read the contents of the 10th slot.
+
+```diff
+-    let bytes_read = host.reveal_dal_page(target_level as i32, 0, 0, &mut buffer)?;
++    let bytes_read = host.reveal_dal_page(target_level as i32, 10, 0, &mut buffer)?;
+```
+
+Recompile the kernel, and deploy it to Weeklynet. This time, it should not be
+able to fetch the content of any slot, because no one is producing them.
+
+#### Hello, World!
+
+The DAL node exposes a RPC just for that: `POST /slots`, whose body is the
```

Comment:
Beware, the RPC is called `/slots` here, but `/slot` in the example below. Which one is right?

Reply:
https://github.com/trilitech/tezos-developer-docs/pull/264/commits/28ac52563ec512ab334a638f4fb799bcc92db959